src/storage/src/hummock/sstable/builder.rs

```diff
@@ -225,11 +234,16 @@ impl<W: SstableWriter> SstableBuilder<W> {
     /// ```
     pub async fn finish(mut self) -> HummockResult<SstableBuilderOutput<W::Output>> {
         let smallest_key = self.block_metas[0].smallest_key.clone();
-        let largest_key = self.last_full_key.clone();
+        let mut largest_key = self.last_full_key.clone();
 
         self.build_block().await?;
         let meta_offset = self.writer.data_len() as u64;
         assert!(!smallest_key.is_empty());
+        for (_, end_user_key) in &self.range_tombstones {
+            if end_user_key.as_slice().gt(user_key(&largest_key)) {
+                largest_key = key_with_epoch(prev_key(end_user_key).to_vec(), 0);
```

Comment:
what about smallest key?

Reply:
🤔 good catch. I also need to update smallest key. Let me fix it.

Reply:
fixed.